src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs

```diff
@@ -33,9 +33,9 @@ public NLogMessageParameterList(IReadOnlyList<KeyValuePair<string, object>> para
         }
 
         /// <summary>
-        /// Attempts to parse the input parameterList
+        /// Create a <see cref="NLogMessageParameterList"/> if <paramref name="parameterList"/> has values, otherwise <c>null</c>
         /// </summary>
-        public static NLogMessageParameterList TryParseList(IReadOnlyList<KeyValuePair<string, object>> parameterList)
+        public static NLogMessageParameterList CreateIfNeeded(IReadOnlyList<KeyValuePair<string, object>> parameterList)
```

Comment:
note: it won't parse IMO, so renamed it to create. Could be also "LazyCreate"

Reply:
Well the NLogMessageParameterList-constructor initiates all the parsing/scanning. And the static-method calls the constructor, so I think TryParse is good.

Reply:
If you rename back to TryParseList, then I will merge.

Reply:
Done, added a `<remarks>` for your remark

Reply:
oops, not proper pushed :(

Reply:
https://github.com/snakefoot/NLog.Extensions.Logging/pull/2